stream_provider.go

```diff
@@ -193,13 +193,17 @@ func (p *StreamProvider) sendLoop(strm grpc.ServerStream, peer Peer) {
 	p.metrics.clientCounter.Inc()
 	broadcaster := p.broadcaster
 	streamCh := make(chan interface{}, p.config.SubscriberInputBufferLen)
-	broadcaster.Register(streamCh, func(config *mux.ConsumerConfig) error {
+	err := broadcaster.Register(streamCh, func(config *mux.ConsumerConfig) error {
 		config.OnBackpressure(func(interface{}) {
 			p.config.OnBackPressure(streamName)
 			p.metrics.backPressureCounter.Inc()
 		})
 		return nil
 	})
+	if err != nil {
+		// errors in broadcaster.Register means one option cannot be applied, this means something is really wrong
+		Log.Fatal("failed to register to broadcaster", zap.Error(err))
```

Comment:
Same comment as in  getAndWatch_stream_provider.go

Reply:
Same as above